src/Microsoft.DotNet.CodeFormatter.Analyzers/ProvideExplicitVariableTypeAnalyzer.cs

```diff
@@ -101,9 +101,13 @@ private static bool IsAnonymousType(SyntaxNode node, SemanticModel model, Cancel
         //       which is (presumbly) much slower. 
         private static bool IsTypeObvious(SyntaxNode node)
         {            
-            if (node == null)
+            // Since parser would generate the missing node (IsMissing == true) and attach a diagnostic to it,
+            // we traverse the sub tree only if a diagnostic is attached as optimization
+            if (node == null || 
+               (node.ContainsDiagnostics && 
+                node.DescendantNodesAndTokensAndSelf().Where(n => n.IsMissing).Any()))
```

Comment:
I am sure we want to minimize node descendants traversal. I suggest you rewrite this code a bit to defer this call until later in the method, when you've conclusively determined you would otherwise return false. Make sense? IOW, eliminate these checks here. Evaluate the rest of the function. If retval appears to be false, put in this logic to sanity check you don't need to flip the result to true due to the missing node case you've identified.